File: backend/src/main/java/es/dawgrupo2/zendashop/extendedDTO/OrderExtendedDTO.java
package es.dawgrupo2.zendashop.extendedDTO;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

import es.dawgrupo2.zendashop.basicDTO.UserBasicDTO;

public record OrderExtendedDTO(
    Long id,
    LocalDateTime creationDate,
    Boolean completed,
    BigDecimal totalPrice,
    BigDecimal shippingCost,
    BigDecimal subtotal,
    UserBasicDTO user,
    String deliveryAddress,
    LocalDate deliveryDate,
    String deliveryNote
) {
    
}

File: frontend/app/dtos/OrderExtendedDTO.ts
import type UserBasicDTO from "./UserBasicDTO";

export default interface OrderExtendedDTO {
  id: number;
  creationDate: string;
  completed: boolean;
  totalPrice: number;
  shippingCost: number;
  subtotal: number;
  user: UserBasicDTO;
  deliveryAddress: string;
  deliveryDate: string;
  deliveryNote: string;
}
